src/common/campaign/source.ts

```diff
@@ -47,13 +49,40 @@ export const validateSquadBoostPermissions = async (
   return source;
 };
 
+export const getSourceTags = async (
+  con: ConnectionManager,
+  sourceId: string,
+): Promise<string[]> => {
+  const result = await con.getRepository(Post).query<{ tag: string }[]>(
+    `
+      WITH recent_posts AS (
+        SELECT id, "sharedPostId"
+        FROM post
+        WHERE "sourceId" = $1
+        ORDER BY "createdAt" DESC
+        LIMIT 30
+      )
+      SELECT DISTINCT ps.keyword AS tag
+      FROM post_keyword ps
+      INNER JOIN recent_posts rp
+        ON ps."postId" = rp.id
+        OR ps."postId" = rp."sharedPostId"
+      LIMIT 30;
+    `,
+    [sourceId],
+  );
+
+  return result.map(({ tag }) => tag.trim()).filter((tag) => tag.length > 0);
```

Comment:
```suggestion
  return result.map(({ tag }) => tag.trim()).filter(Boolean);
```

Reply:
Right. Let me check how this behaves on empty string 👀

Reply:
Worked! Nice!